### PR TITLE
[GEOS-10033] Update startup and shutdown scripts to handle spaces in paths

### PR DESCRIPTION
**Pull request by mwerlen**
[GEOS-10033](https://osgeo-org.atlassian.net/browse/GEOS-10033) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

On linux, I tried to launch a fresh copy of Geoserver from a path containing spaces. It failed.
I looked at the startup.sh and shutdown.sh scripts and fixed quoting issues then linted those files [shellcheck](https://www.shellcheck.net/) and fixed warning.
I hope it will ease the onboarding process for new users :)

[JIRA ticket 10033](https://osgeo-org.atlassian.net/browse/GEOS-10033#)

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] ~~Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)~~
- [ ] ~~Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.~~

## Details

To handle spaces in paths, `exec` command needs to be called with each argument as a single string (quoted if necessary). If an argument is empty, `java` will throw an error. I handle this by replacing empty arguments (`JAVA_OPTS` for example) by useless variables definitions (`-DNoJavaOpts`)

### CODE REVIEW

File: src/release/bin/startup.sh [hunk 1]
```diff
@@ -27,56 +31,59 @@ else
...
+    if [ -r "${GEOSERVER_HOME}/data_dir" ]; then
+        export GEOSERVER_DATA_DIR="${GEOSERVER_HOME}/data_dir"
     else
         echo "No GEOSERVER_DATA_DIR found, using application defaults"
 	      GEOSERVER_DATA_DIR=""
     fi
 fi
 
-cd "$GEOSERVER_HOME"
+cd "${GEOSERVER_HOME}" || exit
```
Comment (randomorder) [inline src/release/bin/startup.sh:77]: if meant to signal a failure may be better to explicitly pass a >0 argument
```suggestion
cd "${GEOSERVER_HOME}" || exit 1
```

File: src/release/bin/shutdown.sh [hunk 2]
```diff
@@ -27,37 +31,38 @@ else
...
-if [ -z $GEOSERVER_HOME ]; then
+if [ -z "${GEOSERVER_HOME:-}" ]; then
   #If GEOSERVER_HOME not set then guess a few locations before giving
   # up and demanding user set it.
   if [ -r start.jar ]; then
      echo "GEOSERVER_HOME environment variable not found, using current "
      echo "directory.  If not set then running this script from other "
      echo "directories will not work in the future."
-     export GEOSERVER_HOME=.
+     GEOSERVER_HOME=$(pwd)
```
Comment (randomorder) [inline src/release/bin/shutdown.sh:41]:
```suggestion
     GEOSERVER_HOME="$(pwd)"
```

File: src/release/bin/startup.sh [hunk 3]
```diff
@@ -27,56 +31,59 @@ else
...
-if [ -z $GEOSERVER_HOME ]; then
+if [ -z "${GEOSERVER_HOME:-}" ]; then
   #If GEOSERVER_HOME not set then guess a few locations before giving
   # up and demanding user set it.
   if [ -r start.jar ]; then
      echo "GEOSERVER_HOME environment variable not found, using current "
      echo "directory.  If not set then running this script from other "
      echo "directories will not work in the future."
-     export GEOSERVER_HOME=`pwd`
+     GEOSERVER_HOME=$(pwd)
```
Comment (randomorder) [inline src/release/bin/startup.sh:41]:
```suggestion
     GEOSERVER_HOME="$(pwd)"
```

File: src/release/bin/shutdown.sh [hunk 4]
```diff
@@ -27,37 +31,38 @@ else
...
-if [ ! -r "$GEOSERVER_HOME"/start.jar ]; then
+if [ ! -r "${GEOSERVER_HOME}/start.jar" ]; then
   echo "The GEOSERVER_HOME environment variable is not defined correctly"
   echo "This environment variable is needed to run this program"
   exit 1
 fi
 
-cd "$GEOSERVER_HOME"
-exec "$_RUNJAVA" $JAVA_OPTS -DSTOP.PORT=8079 -DSTOP.KEY=geoserver -jar start.jar --stop
+cd "${GEOSERVER_HOME}" || exit
```
Comment (randomorder) [inline src/release/bin/shutdown.sh:67]: if meant to signal a failure may be better to explicitly pass a >0 argument
```suggestion
cd "${GEOSERVER_HOME}" || exit 1
```

**Review by randomorder — COMMENTED** [review]
looks good thanks
just a couple minor suggestions

**Review by mprins — CHANGES_REQUESTED** [review]
Needs a Jira ticket so it shows up in the release notes

Tested with succes on OpenBSD/ksh

File: src/release/bin/startup.sh (hunk 1 above)
Comment (mwerlen) [reply src/release/bin/startup.sh]: Done

File: src/release/bin/startup.sh (hunk 3 above)
Comment (mwerlen) [reply src/release/bin/startup.sh]: Done

File: src/release/bin/shutdown.sh (hunk 4 above)
Comment (mwerlen) [reply src/release/bin/shutdown.sh]: Done

File: src/release/bin/shutdown.sh (hunk 2 above)
Comment (mwerlen) [reply src/release/bin/shutdown.sh]: Done

**Comment by mwerlen** [comment]
Thanks for your review @mprins. All requested changes have been applied.
[Jira Ticket](https://osgeo-org.atlassian.net/browse/GEOS-10033#) opened and link added in description.

**Comment by aaime** [comment]
Thanks all, merged. @maddenp up to you if you want to backport to get it in releases sooner, or wait for 2.20-RC release in September.

**Comment by aaime** [comment]
@mwerlen the change might have caused a regression when trying to pass multiple JAVA_OPTS, could you have a look at this:
https://osgeo-org.atlassian.net/browse/GEOS-10316

**Pull request merged**